### PR TITLE
Widgets - Simple Payments: fix a few issues

### PR DESCRIPTION
**Pull request by eliorivero**
Follow-up to #9859 

#### Changes proposed in this Pull Request:

- fix syntax error `$errors.add( 'illegal_params' );`
- add missing message parameter in `->add`
- add missing `jetpack` translation domain
- replace undefined functions `require_lib` and `tracks_record_event`

#### Testing instructions

Test with PHP 5.2 and ensure the widget works normally.

#### Proposed changelog entry for your changes:

PHP 5.2 Compatibility for Simple Payments Widget

### CODE REVIEW

**Comment by oskosk** [comment]
@eliorivero  I Merged Mike's PR and rebased this one

File: modules/widgets/simple-payments.php [hunk 1]
```diff
@@ -388,8 +388,8 @@ private function record_event( $stat_name, $event_action, $event_properties = ar
 
 			// `bumps_stats_extra` only exists on .com
 			if ( function_exists( 'bump_stats_extras' ) ) {
-				require_lib( 'tracks/client' );
-				tracks_record_event( $current_user, 'simple_payments_button_' . $event_action, $event_properties );
+				jetpack_require_lib( 'tracks/client' );
+				jetpack_tracks_record_event( $current_user, 'simple_payments_button_' . $event_action, $event_properties );
```
Comment (rodrigoi) [inline modules/widgets/simple-payments.php]: this block of code is for WordPress.com, where `jetpack_*` functions are not available.

Comment (eliorivero) [reply modules/widgets/simple-payments.php]: Updated 👍

File: modules/widgets/simple-payments/form.php [hunk 2]
```diff
@@ -96,7 +96,7 @@ class="field-currency widefat jetpack-simple-payments-form-product-currency"
 			name="<?php echo esc_attr( $this->get_field_name( 'form_product_currency' ) ); ?>">
 			<?php foreach( Jetpack_Simple_Payments_Widget::$supported_currency_list as $code => $currency ) {?>
 				<option value="<?php echo esc_attr( $code ) ?>"<?php selected( $instance['form_product_currency'], $code ); ?>>
-					<?php esc_html_e( $code . ' ' . $currency ) ?>
+					<?php echo esc_html( "$code $currency" ); ?>
```
Comment (eliorivero) [inline modules/widgets/simple-payments/form.php]: Replacing this here because `esc_html_e` is meant to be used when translating not for all echo.

**Comment by gwwar** [comment]
@oskosk any idea on the timeline when this could be merged? I believe we'll want this in before we land on wpcom for fusion to sync properly.

**Pull request merged**